ci/lava/conftest.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Pytest configuration file."""
+
+import pytest
+import re
+import subprocess
+import sys
+
+
+def pytest_report_teststatus(report):
+    """Override Pytest hook to report test status."""
+    if report.when == "call":
+        if report.passed:
+            lava = "<LAVA_SIGNAL_TESTCASE TEST_CASE_ID={} RESULT=pass>".format(
+                report.nodeid.replace(" ", "_")
+            )
+            return report.outcome, "*", lava
+
+    if report.failed:
+        lava = "<LAVA_SIGNAL_TESTCASE TEST_CASE_ID={} RESULT=fail>".format(
+            report.nodeid.replace(" ", "_")
+        )
+        return report.outcome, "*", lava
+
+
+def pytest_addoption(parser):
+    """Add option parsing to pytest."""
+    parser.addoption("--debug_output", action="store_true")
+    parser.addoption("--device", action="store", default="none")
+    parser.addoption("--dut", action="store", default="auto")
+    parser.addoption(
+        "--dut_download_dir", action="store", default="/tmp/pytest"
+    )
+    parser.addoption("--dut_tutorials_dir", action="store", default="/scratch")
+    parser.addoption(
+        "--host_download_dir", action="store", default="/tmp/pytest"
+    )
+    parser.addoption(
+        "--host_tutorials_dir", action="store", default="/tmp/tutorials"
+    )
+    parser.addoption("--venv", action="store", default="/tmp/venv")
+
+
+@pytest.fixture
+def debug_output(request):
+    """Fixture for --debug_output."""
+    return request.config.getoption("--debug_output")
+
+
+@pytest.fixture
+def device(request):
+    """Fixture for --device."""
+    return request.config.getoption("--device")
+
+
+@pytest.fixture
+def dut(request):
+    """Fixture for --dut."""
+    return request.config.getoption("--dut")
+
+
+@pytest.fixture
+def dut_download_dir(request):
+    """Fixture for --dut_download_dir."""
+    return request.config.getoption("--dut_download_dir")
+
+
+@pytest.fixture
+def dut_tutorials_dir(request):
+    """Fixture for --dut_tutorials_dir."""
+    return request.config.getoption("--dut_tutorials_dir")
+
+
+@pytest.fixture
+def host_download_dir(request):
+    """Fixture for --host_download_dir."""
+    return request.config.getoption("--host_download_dir")
+
+
+@pytest.fixture
+def host_tutorials_dir(request):
+    """Fixture for --host_tutorials_dir."""
+    return request.config.getoption("--host_tutorials_dir")
+
+
+@pytest.fixture
+def venv(request):
+    """Fixture for --venv."""
+    return request.config.getoption("--venv")
+
+
+class Execute_Helper:
```

Comment:
Convention is to not use underscores in class names. Should be `ExecuteHelper`, same comment for other classes.

Reply:
changed